autoload/codefmt.vim

```diff
@@ -107,20 +107,40 @@ endfunction
 " @function(#SetWhetherToPerformIsAvailableChecksForTesting), skips the
 " IsAvailable check and always returns true.
 function! s:IsAvailable(formatter) abort
-  if codefmt#ShouldPerformIsAvailableChecks()
-    return a:formatter.IsAvailable()
+  if !codefmt#ShouldPerformIsAvailableChecks()
+    return 1
   endif
-  return 1
+  return a:formatter.IsAvailable()
+endfunction
+
+
+" Checks whether {formatter} is available, safely handling errors by logging
+" an error and returning 0.
+function! s:IsAvailableSafe(formatter) abort
+  try
+    return s:IsAvailable(a:formatter)
+  catch /.*/
+    call maktaba#error#Shout(
+        \ 'Failed to evaluate whether formatter %s is available: %s',
+        \ a:formatter.name,
+        \ v:exception)
+    return 0
+  endtry
 endfunction
 
 
 ""
 " Detects whether a formatter has been defined for the current buffer/filetype.
 function! codefmt#IsFormatterAvailable() abort
-  let l:formatters = copy(s:registry.GetExtensions())
-  let l:is_available = 'v:val.AppliesToBuffer() && s:IsAvailable(v:val)'
-  return !empty(filter(l:formatters, l:is_available)) ||
-      \ !empty(get(b:, 'codefmt_formatter'))
+  if !empty(get(b:, 'codefmt_formatter'))
+    return 1
+  endfor
```

Comment:
Wut. I somehow used `:endfor` instead of `:endif` here without seeing any test failures. Gotta fix that and check what's up with test coverage…

Reply:
Oh, because the tests usually disable it with `#SetWhetherToPerformIsAvailableChecksForTesting()`. :facepalm: 

Stay tuned for fix.

Reply:
Fixed in 9346c381c.